File: web_ui/app/features/authentication/domain/usecases/login_user.usecase.ts
import { LoginUserDto } from "../../data/dto/LoginUser.dto";
import { loginUser } from "../../data/redux/authSlice";

export class LoginUserUsecase {
  async execute(
    {
      userName,
      password,
    }: {
      userName: string;
      password: string;
    },
    dispatch: any
  ): Promise<void> {
    const loginUserDto = new LoginUserDto(userName, password);
    dispatch(loginUser(loginUserDto));
  }
}

File: mobile_ui/quickchat_ui/lib/features/authentication/usecase/login_user.usecase.dart
import 'package:injectable/injectable.dart';
import 'package:quickchat_ui/features/authentication/data/bloc/auth_bloc.dart';
import 'package:quickchat_ui/features/authentication/data/dto/LoginUser.dto.dart';

@injectable
class LoginUserUsecase {
  LoginUserUsecase({required this.authBloc});

  final AuthBloc authBloc;

  void execute(LoginUserDto req) {
    authBloc.add(AuthEvent.loginUser(loginUserDto: req));
  }
}
